procfs: add total time running time of task

Gary-Hobson opened the pull request:
## Summary
add total time running time of task

```sh
  nsh> critmon
  PRE-EMPTION   CSECTION      RUN         TIME         PID   DESCRIPTION
  0.010265000   0.000037000   ----------- -----        ----  CPU 0
  0.000000000   0.000000000   0.001237000 28.421047000 0     Idle Task
  0.000011000   0.000037000   0.000046000 0.034211000  1     loop_task
  0.000000000   0.000028000   0.000067000 0.236657000  2     hpwork
```
## Impact

## Testing

acassis commented:
@Gary-Hobson please update the Documentation/applications/nsh/commands.rst "procfs NOTE:" comment, adding about this new features.

Gary-Hobson commented:
> please update the Documentation/applications/nsh/commands.rst "procfs NOTE:" comment, adding about this new features.

I moved it into proc_critmon and added documentation for the critmon command

pkarashchenko commented on fs/procfs/fs_procfsproc.c:
```diff
@@ -854,12 +855,27 @@ static ssize_t proc_critmon(FAR struct proc_file_s *procfile,
+
+  up_perf_convert(tcb->run_time, &runtime);
+
+  linesize = procfs_snprintf(procfile->line, STATUS_LINELEN,
+                            "%lu.%09lu\n",
+                            (unsigned long)runtime.tv_sec,
+                            (unsigned long)(runtime.tv_nsec));
```
```suggestion
  linesize = procfs_snprintf(procfile->line, STATUS_LINELEN,
                             "%lu.%09lu\n",
                             (unsigned long)runtime.tv_sec,
                             (unsigned long)runtime.tv_nsec);
```

Gary-Hobson replied:
done

pkarashchenko commented on fs/procfs/fs_procfsproc.c:
```diff
@@ -762,6 +762,7 @@ static ssize_t proc_critmon(FAR struct proc_file_s *procfile,
                             size_t buflen, off_t offset)
 {
   struct timespec maxtime;
```
why not rename `maxtime` to `ts` and reuse it instead of adding new stack variable?

xiaoxiang781216 commented on fs/procfs/fs_procfsproc.c:
```diff
@@ -854,9 +854,24 @@ static ssize_t proc_critmon(FAR struct proc_file_s *procfile,
+
+  up_perf_convert(tcb->run_time, &ts);
+
+  linesize = procfs_snprintf(procfile->line, STATUS_LINELEN,
```
merge to line 860

Gary-Hobson replied:
done

xiaoxiang781216 commented:
@Gary-Hobson please rebase your change.
